app/controllers/orders_controller.rb

```diff
@@ -48,28 +80,89 @@ def new
   def update
     @order = Order.find(session[:order_id]) #session is persistent, the cookie has the session information.
     @order.update(order_param[:order]) # when you get a request, here is my information for right now.
+
     reduce_inventory(@order)
-    if @order.save
+
+    zip = params["order"]["zip"]
+    city = params["order"]["city"]
+    state = params["order"]["state"]
+    # message = ""
+    message = valid_location(zip, city, state)
+
+
+    if @order.save && message == ""
       redirect_to order_path
```

Comment:
Rails has a handy helper method you could use here:
message.blank?

It would match if message were "   " or nil, etc.